src/event.js

```diff
@@ -554,9 +554,13 @@ function leverageNative( el, type, expectSync ) {
 			if ( ( event.isTrigger & 1 ) && this[ type ] ) {
 
 				// Interrupt processing of the outer synthetic .trigger()ed event
-				if ( !saved ) {
+				// Saved data should be false in such cases, but might be a leftover capture object
+				// from an async native handler (gh-4350)
+				if ( !saved.length ) {
```

Comment:
wouldn't this break in case `saved` is `false` (set in `dataPriv.set( this, type, false );` on line 570) ?

Reply:
@stof No, it works:
```js
var b = false;
!b.length; // true
```
This is because for a boolean `b` `b.length` will aut-box `b` into a `Boolean` instance which doesn't have the `length` field so `b.length` will be `undefined`. A negation of `undefined` is `true`.

I know it's not the most readable code but we care about size and this behavior is covered by tests.